### ISSUE TITLE
Test failures - first test run only (1.8.2)

### ISSUE THREAD

**Issue by knweiss** [prometheus/prometheus#3436]
**What did you do?**

(I think I ran into this with 1.8.1 as well but I'm not 100% sure). This is with
1.8.2: The first time `make test` runs after upgrading the prometheus git repo I see the following test failures:

```
go/src/github.com/prometheus/prometheus (release-1.8) $ git describe --tag
v1.8.2
go/src/github.com/prometheus/prometheus (release-1.8) $ make
>> formatting code
>> running staticcheck
>> fetching promu
>> building binaries
 >   prometheus
 >   promtool
>> running all tests
ok      github.com/prometheus/prometheus/cmd/prometheus 0.066s
?       github.com/prometheus/prometheus/cmd/promtool   [no test files]
ok      github.com/prometheus/prometheus/config 0.025s
ok      github.com/prometheus/prometheus/discovery      0.060s
?       github.com/prometheus/prometheus/discovery/azure        [no test files]
ok      github.com/prometheus/prometheus/discovery/consul       0.018s
?       github.com/prometheus/prometheus/discovery/dns  [no test files]
?       github.com/prometheus/prometheus/discovery/ec2  [no test files]
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": unexpected end of JSON input" source="file.go:200"
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_invalid_nil.json": unexpected end of JSON input" source="file.go:200"
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_invalid_nil.json": nil target group item found" source="file.go:200" 
time="2017-11-08T13:55:13+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_invalid_nil.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_invalid_nil.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_invalid_nil.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_invalid_nil.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:28+01:00" level=error msg="Error reading file "fixtures/_test_valid.json": invalid character ']' looking for beginning of value" source="file.go:200"
time="2017-11-08T13:55:29+01:00" level=error msg="Error reading file "fixtures/_test_valid.yml": yaml: did not find expected node content" source="file.go:200"
time="2017-11-08T13:55:29+01:00" level=error msg="Error reading file "fixtures/_test_valid.yml": yaml: did not find expected node content" source="file.go:200"
--- FAIL: TestFileSD (16.61s)
        file_test.go:95: Unexpected target groups [], we expected a failure here.
FAIL
FAIL    github.com/prometheus/prometheus/discovery/file 16.613s
?       github.com/prometheus/prometheus/discovery/gce  [no test files]
ok      github.com/prometheus/prometheus/discovery/kubernetes   0.099s
[...]
```

**What did you expect to see?**

No test failures.

**What did you see instead? Under which circumstances?**

Tests fail as shown above.

When I execute a 2nd `make test` all the tests succeed.

This is a very low prio issue. Basically, I just wanted to let you know that I see this.

**Environment**

* System information:
```
$ go version
go version go1.9.2 linux/amd64
$ uname -srm
Linux 3.10.0-693.5.2.el7.x86_64 x86_64
```

* Prometheus version:

1.8.2

**Comment by knweiss** [comment]
This could be related to #3419 (which I've just discovered).

**Comment by grobie** [comment]
Fixed in #3438 (for master branch). Reopen if you still see them.
